client/router.js

```diff
@@ -0,0 +1,108 @@
+/*global window NODE_ENV*/
+import queryString from 'query-string';
+import extend from 'extend';
+
+import ServerRouter from '../server/router'
+import { createAction } from 'redux-act';
+
+const updateLocation = createAction('Default action for updating url.');
+
+export { updateLocation };
+
+
+const DEFAULT_UPDATE_LOCATION_ACTION = {
+  type: updateLocation.getType(),
+  payload: {}
+};
+
+export default class Router extends ServerRouter {
+
+  get locale(){
+    return Router.locale();
+  }
+
+  shouldUpdateCurrentRoute(location){
+    let router = this;
+    return !router.current_route || !router.current_route.matchesLocation(location.pathname);
+  }
+
+  initializeHistory(createHistory, store) {
+    let router = this;
+    router.history = createHistory();
+    router.history.listen(router.onLocationChange.bind(router, store));
+  }
+
+  // Changing Route
+
+  // this will cause onLocationChange to fire with
+  // the new location.
+  pushRoute(route_name, action, payload){
+    let router = this,
+        route = router.routes.getRoute(route_name);
+
+    action = {
+      type: action ? action.getType() : updateLocation.getType(),
+      payload: payload,
+      no_scroll: payload ? payload.no_scroll : false
+    };
+
+    router.pushHistory({
+      pathname: route.url(action, router.i18n),
+      state: action
+    });
+  }
+
+  pushHistory(location){
+    this.history.push(location);
+  }
+
+  onLocationChange(store, new_location){
+    console.log("Location changed!");
+    let router = this;
+
+    if (new_location.action !== 'PUSH') return false;
```

Comment:
Actually, deprecate this line. We need to allow push actions so user can use back and forward navigation buttons in their browser.